core/src/validator.rs

```diff
@@ -197,6 +196,14 @@ pub struct Validator {
     ip_echo_server: solana_net_utils::IpEchoServer,
 }
 
+fn abort() -> ! {
+    #[cfg(not(test))]
+    std::process::exit(1);
+
+    #[cfg(test)]
+    panic!("process::exit(1) is intercepted for friendly test failure...");
```

Comment:
nice idea

Reply:
+1

Reply:
thanks! Ideally, I want to get rid of all `process:exit(1)`s in lib code... But, I don't have enough bandwidth; this is a tiny 10-ish lines of hacky work-around for the current situation...